boards/build_scripts/tock_kernel_layout.ld

```diff
@@ -43,33 +43,14 @@ PAGE_SIZE = DEFINED(PAGE_SIZE) ? PAGE_SIZE : 512;
 
 SECTIONS
 {
-   .stack (NOLOAD) :
-    {
-        /* Kernel stack.
-         *
-         * Tock places the kernel stack at the bottom of SRAM so that the
-         * kernel will trigger memory fault if it exceeds its stack depth,
-         * rather than silently overwriting valuable data.
-         */
-        . = ALIGN(8);
-         _sstack = .;
-
-         /* For GNU LD, we can just advance the location pointer (".") here to
-          * reserve space for the stack. That, however, doesn't seem to work
-          * for LLVM LLD. The resulting ELF has a stack section that shows the
-          * correct size, but the next section (in our case .relocate) is not
-          * moved down as well, instead it sits at the same address as .stack.
-          * To work around this, we declare a dummy buffer and then insert it
-          * here in the .stack section. This sets the stack size correctly and
-          * places the .relocate section at the correct address. */
-         KEEP(*(.stack_buffer))
-         /*. = . + 0x1000;*/  /*This is the original method. */
-
-         . = ALIGN(8);
-         _estack = .;
-    } > ram
-
 
+    
+    /**
+     * The order matters here for the x86 platform, the .text section has to be the first one.
+     * Please see the details bellow where the .x86.start is stored.
+     */
```

Comment:
I don't think this necessary, or maybe is perhaps confusing in its current form. The order in `.text` matters for virtually all platforms: Cortex-M expects the vector table to be located at the start, RISC-V requires the `_start` symbol to be first, and same for `.x86.start`.

So I'd say we can either add a general "the order matters!" comment, or we just leave it as is. Each one of the first three entries in `.text` clearly documents that it needs to go first for its respective architecture.

Reply:
```suggestion
    /**
     * The order that section contents ends up emitted in the final physical ELF file is
     * significant for systems like QEMU that use multiboot. Only the first 8192 bytes of the file
     * are scanned ( https://www.gnu.org/software/grub/manual/multiboot/multiboot.html ).
     *
     * For this reason, it is important that the `.text` section, which contains the `.multiboot`
     * section, is defined first in this file as tools tend to write sections out in linear order.
     */
```